api/foxpass_radius_logs.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/local/bin/python3
+
+"""
+This script pulls RADIUS logs from your Foxpass instance and allows you to parse them by date, user, IP, or connection outcome. 
+Logs can be printed (in pretty colors) or exported in CSV format. 
+
+Required packages:
+pip install requests
+
+To run:
+python foxpass_radius_logs.py
+
+By default the script will print color-coded RADIUS logs from the last (5) days. You can use the optional arguments below:
+
+--hours - How far back to show the logs in Hours.
+--user - Filter by user.
+--location - Filter by RADIUS Client, based on the items defined in the OFFICE_IPS dict.
+--outcome - Filter by outcome of the connection, specify True or False.
+--csv - Output the logs to a CSV file, specify the filename and path.
+"""
+
+from datetime import datetime, timedelta, timezone
+import requests
+import argparse
+import csv
+
+##### EDIT THESE #####
+FOXPASS_API_TOKEN = ""
+# RADIUS clients, can be called with the --location argument.
+OFFICE_IPS = { 
+    "office1":"", 
+    "office2":"", 
+    "office3":"", 
+    }
+
+# "YYYY-MM-DDTHH:MMZ". STARTDATE Default 5 days ago, ENDDATE is current day/time in UTC. Can be changed with the --hours argument.
+STARTDATE = (datetime.now(timezone.utc) - timedelta(days=5)).strftime('%Y-%m-%dT%H:%MZ')
+ENDDATE = datetime.now(timezone.utc).strftime('%Y-%m-%dT%H:%MZ')
+
+FOXPASS_URL = "https://api.foxpass.com/v1/logs/radius/"
+HEADERS = {'Authorization': 'Token ' + FOXPASS_API_TOKEN}
+PAGEREQUEST = requests.post(FOXPASS_URL, json={"from": STARTDATE, "to": ENDDATE}, headers=HEADERS).json()
+PAGES = PAGEREQUEST["numPages"]
+
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKCYAN = '\033[96m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Pull and parse RADIUS logs from your Foxpass environment')
+    parser.add_argument('--user', help='Filter logs by username')
+    parser.add_argument('--outcome', help='Filter logs by connection outcome: True or False')
+    parser.add_argument('--hours', type=int, help='How far back to check the logs in hours')
+    parser.add_argument('--location', help='Filter logs by location: tlv, tlv-backup, nyc, nyc-backup, sha')
+    parser.add_argument('--csv', help='Export a CSV of the log data to the specified filename and path')
+    return parser.parse_args()
+
+# Builds an if statement to filter the logs based on user arguments
+def build_query(username=None, outcome=None, location=None):
+    query_string = ""
+    if username != None:
+        query_string += f"log['username']=='{username}' and "
+    if location != None:
+	    query_string += f"log['ipAddress']=='{location}' and "
+    if outcome != None:
+        query_string += f"log['success']=={outcome}"
+    if query_string != "":
+        if query_string[-2] == 'd':
+            query_string = query_string[:-5]
```

Comment:
is this just stripping the trailing ` and `?

Reply:
You got it. It's just making sure that if the string ends with "and" that it gets stripped off. I'll add a comment.